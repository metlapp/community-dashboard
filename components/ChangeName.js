import React, { useContext, useEffect } from "react";
import { Button, Title, TextInput } from "react-native-paper";
import AuthContext from "../auth/Context";
import authStorage from "../auth/Storage";
import axios from "axios";
import { apiConfig } from "../config/config";
import PropTypes from "prop-types";

const ChangeName = (props) => {
  const authContext = useContext(AuthContext);
  var newName = "";
  const [error, setError] = React.useState(false);

  // Posts new name to the API
  const postName = async () => {
    axios
      .patch(
        apiConfig.baseUrl + "users/" + authContext.user.id + "/",
        { first_name: newName },
        {
          auth: apiConfig.auth,
        }
      )
      .then((data) => {
        authContext.setUser(data.data);
        authStorage.storeUser(data.data);
      })
      .catch(console.error);
  };

  const saveAndClose = () => {
    newName = newName.trim();
    if (newName == "") {
      setError(true);
    } else {
      postName();
      setError(false);

      props.hidemodal();
    }
  };

  return (
    <>
      <Title>Hello, {authContext.user.first_name}</Title>
      <TextInput
        className="newNameInput"
        placeholder="Change Name"
        onChangeText={(text) => (newName = text)}
        error={error}
      />
      <Button className="save" onPress={saveAndClose}>
        Save
      </Button>
    </>
  );
};

ChangeName.propTypes = {
  hidemodal: PropTypes.func,
};

export default ChangeName;
