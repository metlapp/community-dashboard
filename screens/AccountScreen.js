import React from "react";
import { SafeAreaView, StyleSheet, View } from "react-native";
import { Button, Appbar } from "react-native-paper";
import AccountModal from "../components/AccountModal";

const AccountScreen = (props) => {
  //determines what component the modal will render when a button is clicked
  const [form, setForm] = React.useState("");
  const [visible, setVisible] = React.useState(false);
  const showModal = () => setVisible(true);
  const hideModal = () => setVisible(false);
  return (
    <SafeAreaView>
      <Appbar.Header>
        <Appbar.Content title="ACCOUNT" />
      </Appbar.Header>
      {/* Opens model and takes in the visible state with what component to render */}
      <AccountModal
        visibility={visible}
        hidemodal={hideModal}
        form={form}
        {...props}
      />
      <View style={styles.buttonContainer}>
        {/* When Button is clicked it changes the state of the visibility and
           tells the modal what component to render inside the modal */}
        <Button
          className="button"
          style={styles.buttonStyle}
          icon="email"
          mode="contained"
          onPress={() => {
            setForm("Email");
            setVisible(true);
          }}
        >
          Change Email address
        </Button>
        <Button
          className="button"
          style={styles.buttonStyle}
          icon="account-edit"
          mode="contained"
          onPress={() => {
            setForm("Name");
            showModal();
          }}
        >
          Change name
        </Button>
        <Button
          className="button"
          style={styles.buttonStyle}
          icon="square-edit-outline"
          mode="contained"
          onPress={() => {
            setForm("Password");
            showModal();
          }}
        >
          Change password
        </Button>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  title: {
    textAlign: "center",
  },
  buttonContainer: {
    marginTop: "50%",
    alignItems: "center",
    justifyContent: "center",
  },
  buttonStyle: {
    margin: 10,
    width: "100%",
    height: 60,
    justifyContent: "center",
  },
});

export default AccountScreen;
