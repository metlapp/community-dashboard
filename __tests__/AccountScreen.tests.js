import "react-native";
import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import AccountScreen from "../screens/AccountScreen";
import ChangeName from "../components/ChangeName";
import { render, fireEvent } from "@testing-library/react-native";
// This test just uses Jest snapshot testing
it("renders correctly, test using Jest", () => {
  renderer.create(<AccountScreen />);
});

// Using Jest + Enzyme
describe("<AccountScreen />", () => {
  it("renders correctly, test using Jest + Enzyme", () => {
    expect(shallow(<AccountScreen />)).toMatchSnapshot();
  });
  it("matches the snapshot", () => {
    const tree = renderer.create(<AccountScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should have three buttons", () => {
    const component = shallow(<AccountScreen />);
    const wrapper = component.find(".button");
    expect(wrapper.getElements()).toHaveLength(3);
  });
});
