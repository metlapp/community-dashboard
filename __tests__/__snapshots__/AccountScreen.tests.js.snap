// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountScreen /> matches the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "backgroundColor": "#6200ee",
          "elevation": 4,
          "zIndex": 0,
        },
        Object {
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#6200ee",
          "elevation": 0,
          "flexDirection": "row",
          "height": 56,
          "marginTop": 0,
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          Object {
            "width": 48,
          }
        }
      />
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "flex": 1,
              "paddingHorizontal": 12,
            },
            Array [
              false,
              Object {
                "alignItems": "center",
              },
              undefined,
            ],
          ]
        }
      >
        <Text
          accessibilityRole="header"
          accessibilityTraits="header"
          accessible={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "fontSize": 17,
                },
                undefined,
              ],
            ]
          }
        >
          ACCOUNT
        </Text>
      </View>
      <View
        style={
          Object {
            "width": 48,
          }
        }
      />
    </View>
  </RCTSafeAreaView>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": "50%",
      }
    }
  >
    <View
      className="button"
      style={
        Object {
          "backgroundColor": "#6200ee",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "height": 60,
          "justifyContent": "center",
          "margin": 10,
          "minWidth": 64,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "marginLeft": 12,
                "marginRight": -4,
              }
            }
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 16,
                  },
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "material-community",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                ],
              ]
            }
          >
            Change Email address
          </Text>
        </View>
      </View>
    </View>
    <View
      className="button"
      style={
        Object {
          "backgroundColor": "#6200ee",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "height": 60,
          "justifyContent": "center",
          "margin": 10,
          "minWidth": 64,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "marginLeft": 12,
                "marginRight": -4,
              }
            }
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 16,
                  },
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "material-community",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                ],
              ]
            }
          >
            Change name
          </Text>
        </View>
      </View>
    </View>
    <View
      className="button"
      style={
        Object {
          "backgroundColor": "#6200ee",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "height": 60,
          "justifyContent": "center",
          "margin": 10,
          "minWidth": 64,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "marginLeft": 12,
                "marginRight": -4,
              }
            }
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 16,
                  },
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "material-community",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              喇
            </Text>
          </View>
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                ],
              ]
            }
          >
            Change password
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`<AccountScreen /> renders correctly, test using Jest + Enzyme 1`] = `
<ForwardRef(SafeAreaView)>
  <withTheme(Appbar.Header)>
    <withTheme(Appbar.Content)
      title="ACCOUNT"
    />
  </withTheme(Appbar.Header)>
  <AccountModal
    form=""
    hidemodal={[Function]}
    visibility={false}
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": "50%",
      }
    }
  >
    <withTheme(Button)
      className="button"
      icon="email"
      mode="contained"
      onPress={[Function]}
      style={
        Object {
          "height": 60,
          "justifyContent": "center",
          "margin": 10,
          "width": "100%",
        }
      }
    >
      Change Email address
    </withTheme(Button)>
    <withTheme(Button)
      className="button"
      icon="account-edit"
      mode="contained"
      onPress={[Function]}
      style={
        Object {
          "height": 60,
          "justifyContent": "center",
          "margin": 10,
          "width": "100%",
        }
      }
    >
      Change name
    </withTheme(Button)>
    <withTheme(Button)
      className="button"
      icon="square-edit-outline"
      mode="contained"
      onPress={[Function]}
      style={
        Object {
          "height": 60,
          "justifyContent": "center",
          "margin": 10,
          "width": "100%",
        }
      }
    >
      Change password
    </withTheme(Button)>
  </View>
</ForwardRef(SafeAreaView)>
`;
